pkgs/development/interpreters/lua-5/default.nix

```
@@ -6,11 +6,7 @@ rec {
     sourceVersion = { major = "5"; minor = "4"; patch = "2"; };
     hash = "0ksj5zpj74n0jkamy3di1p6l10v4gjnd2zjnb453qc6px6bhsmqi";
 
-    patches = if stdenv.isDarwin then [ ./5.4.darwin.patch ] else [
-      # build lua as a shared library as well, MIT-licensed from
-      # https://github.com/archlinux/svntogit-packages/tree/packages/lua/trunk
-      ./liblua.so.patch
-    ];
+    patches = optional stdenv.isDarwin ./5.4.darwin.patch;
```

I don't believe `optional` is in scope here?

```suggestion
    patches = lib.optional stdenv.isDarwin ./5.4.darwin.patch;
```